nil/core-concepts/shards-parallel-execution.mdx

```diff
@@ -29,6 +29,92 @@ As the number of execution shards grows, this process may be changed so that sha
 
 :::
 
+### Async execution
+
+Generally, the term 'async execution' describes delegating the completion of a task to a dedicated CPU thread or a similar isolated component. While this task is processed, the main program can freely execute other instructions, and then react to receiving the results of the async task.
+
+In =nil;, async execution refers to calling a contract that is deployed on another shard. The time taken to execute this call may vary, and the 'caller' contract is also free to execute other instructions without having to wait for the results of the call.
```

Comment:
Async execution doesn't necessarily mean "another shard". It is perfectly fine to make an async call to the same shard. The execution will likely happen in the next block.

Reply:
Yes, you're right. I have rephrased the description